### DISCUSSION TITLE
Add hotkeys for annotation tools.

### DISCUSSION THREAD

**Discussion by XMB-7** [gildas-lormeau/SingleFile#1165] [Ideas]
This would significantly enhance the convenience for heavy annotation users.
For example, set num or alt+num,

![image](https://github.com/gildas-lormeau/SingleFile/assets/65625632/f6003007-d51a-4fa6-9b09-c189cd4d9d5f)

**Comment by gildas-lormeau** [comment]
Thank you for the suggestion. Note that some shortcuts already exists. `num6` and `num7`are respectively `Ctrl-Z` and `Ctrl-Shift-Z`. `num8` is `Ctrl-S` and you can print the page with `Ctrl-P`. I think it is a good idea to implement at least `num1` and `num2`. Maybe the it could combined with `Ctrl`for toggling features.

**Comment by XMB-7** [comment]
Thank you for considering adding shortcuts for annotation tools. In my opinion, `num2` and `num3` are the most helpful ones for the purpose of continuous highlighting. It would be even better if `alt+z` and `alt+shift+z` could be added specifically to undo and redo highlighting operations.

**Comment by TFWol** [comment]
I was just think about this after a 3 hour long session of page annotations.

For Firefox, maybe leverage the "Manage Extension Shortcuts" feature so we can define our own.

Kind of like this:
![image](https://github.com/user-attachments/assets/a62f7b8f-91c0-4626-8d69-d06cf49c6425)


Apparently you can add as many shortcuts as you want. Tree Style Tab has a ton:
![image](https://github.com/user-attachments/assets/e50ead04-bd31-4f94-9937-b479d5e22450)

**Reply by gildas-lormeau** [reply]
Thank you for the suggestion, I had not thought of it. However, these shortcuts are supposed to be usable from anywhere in the browser. The problem here is that they are specific to the editor and could lead to confusion for some user maybe.

**Reply by TFWol** [reply]
Good point. In that case, probably have something like what [Tab Unloader for Tree Style Tab](https://addons.mozilla.org/en-US/firefox/addon/tab-unload-for-tree-style-tab/) does for capturing keybindings:

---
![image](https://github.com/user-attachments/assets/8f7cc25f-65e9-41a2-9f6f-a6c547ce3bf1)
